From a095b07d0b9aa142a8e4eeec00be2cd65ea9ee0c Mon Sep 17 00:00:00 2001
From: "Cao,Bing Bu" <bingbu.cao@intel.com>
Date: Tue, 28 Apr 2015 14:32:30 +0800
Subject: [PATCH 1/2] CSS: workaround to prevent refcount crash

[CHT PV Only]
Sometimes the buffer that dequeued from FW is corrupted,
the refcount for this buffer is broken and invalid, and
then cause crash in refcount.

This patch remove the assert as a workaround.

Change-Id: I4ba4459300eab7aade42d1cadedd8f952184f5c1
Original-Tracked-On: #JIMINAN-34266
Signed-off-by: Cao,Bing Bu <bingbu.cao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/53289
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Approver: Bin B Han <bin.b.han@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: Yunliang Ding <yunliang.ding@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
Tracked-On:https://jira01.devtools.intel.com/browse/OAM-4372
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/458155
---
 .../camera/drivers/media/pci/atomisp2/css/base/refcount/src/refcount.c  | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/base/refcount/src/refcount.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/base/refcount/src/refcount.c
index fc97748..66ce992 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/base/refcount/src/refcount.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/base/refcount/src/refcount.c
@@ -196,7 +196,6 @@ bool ia_css_refcount_decrement(int32_t id, hrt_vaddress ptr)
 			id, ptr, entry, entry->id, entry->count);
 	else
 		IA_CSS_ERROR("entry NULL\n");
-	assert(false);
 
 	return false;
 }
@@ -246,7 +245,6 @@ void ia_css_refcount_clear(int32_t id, clear_func clear_func_ptr)
 						    "no clear_func\n");
 				mmgr_free(entry->data);
 			}
-			assert(entry->count == 0);
 			if (entry->count != 0) {
 				IA_CSS_WARNING("Ref count for entry %x is not zero!", entry->id);
 			}
-- 
1.9.1

